src/Core/src/Handlers/Window/WindowHandler.iOS.cs

```diff
@@ -1,17 +1,37 @@
 ﻿using System;
-using ObjCRuntime;
+using Foundation;
 using UIKit;
 
 namespace Microsoft.Maui.Handlers
 {
 	public partial class WindowHandler : ElementHandler<IWindow, UIWindow>
 	{
+		private IDisposable? _effectiveGeometryObserver;
+
 		protected override void ConnectHandler(UIWindow platformView)
 		{
 			base.ConnectHandler(platformView);
 
 			UpdateVirtualViewFrame(platformView);
+
+			_effectiveGeometryObserver = platformView.WindowScene?.AddObserver("effectiveGeometry", NSKeyValueObservingOptions.OldNew, HandleEffectiveGeometryObserved);
```

Comment:
Referencing the handler directly from the native view - the observer now holds a reference to the handler - will most certainly cause a leak.

Please use our amazing and totally not verbose way to do events: https://github.com/dotnet/maui/blob/main/src/Core/src/Handlers/Switch/SwitchHandler.iOS.cs#L57-L79